src/translib/transformer/xfmr_vrf.go

```diff
@@ -514,4 +523,189 @@ var DbToYang_network_instance_route_distinguisher_field_xfmr KeyXfmrDbToYang = f
         return res_map, err
 }
 
+/* YangToDb subtree transformer for network instance interface binding */
+var YangToDb_network_instance_interface_binding_subtree_xfmr SubTreeXfmrYangToDb = func(inParams XfmrParams) (map[string]map[string]db.Value, error) {
+        var err error
+        res_map := make(map[string]map[string]db.Value)
+
+        log.Infof("YangToDb_network_instance_interface_binding_subtree_xfmr: ygRoot %v uri %v", inParams.ygRoot, inParams.uri)
+
+        pathInfo := NewPathInfo(inParams.uri)
+
+        targetUriPath, err := getYangPathFromUri(pathInfo.Path)
+
+        log.Info("YangToDb_network_instance_interface_binding_subtree_xfmr: targetUri ", targetUriPath)
+
+        /* get the name at the top network-instance table level, this is the key */
+        keyName := pathInfo.Var("name")
+        if (strings.HasPrefix(keyName, "Vrf") == false) {
+                log.Info("YangToDb_network_instance_interface_binding_subtree_xfmr, not Vrf nothing to do with interface", keyName)
+                return res_map, errors.New("Unexpected network instance name")
+        }
+
+        /* Check if interfaces is configured, if not, return */
+        vrfObj := getNwInstRoot(inParams.ygRoot)
+
+        if vrfObj.NetworkInstance[keyName].Interfaces == nil {
+                return res_map, err
+        }
+
+        for intfId, _:= range vrfObj.NetworkInstance[keyName].Interfaces.Interface {
+                intf_type, _, err := getIntfTypeByName(intfId)
+                if err != nil {
+                        log.Info("YangToDb_network_instance_interface_binding_subtree_xfmr: unknown intf type for ", intfId)
+                }
+
+                var intf_tbl_name string
+                intf_tbl_name = intfTblName[intf_type]
```

Comment:
The logic to get the DB Table name from interface name is better to be present xfmr_intf.go. When adding new interface type, it would be better if we have all the related code together.

Reply:
done. use xrf_intf.go, IntfTypeTblMap[intf_type] and getIntfTableNameByDBId